gravitational/next
# Remove variable exports from generated mdx files

I researched issue that @deliaconstantino was reported yesterday about hot reloading not working with mdx pages.

Its a bug that is already fixed in `12.0.9-canary.9` so it will be automatically resolved when we update to `next@12.0.9` .But I don't want to update to `canary` version and we need to work with pages now and not then they release it. So I made a quick fix that resolves this issue for 99% use cases.

So what was the problem:

As the link in the popup says if the page or component file exports something except React components AND if this export is used by non-react code, it causes forced reload. 

We do have autogenerated exports from each mdx page (we export object with frontmatter fields), but they are not imported by any other file, so they shouldn't cause force reload and they did not cause it before. But in the 12.0.8 Next for some reason decided that they should cause reloads. As I wrote before it is fixed in 12.0.9, but for now I removed autogenerated export declarations from mdx pages, because we does not really need them, TBH. So now it works again.

There is one exception that still requires manual reloads – pages with `getStaticProps` like https://raw.githubusercontent.com/gravitational/next/main/pages/resources/index.mdx

They can be fixed too, if we replace imports from `server/resources-helpers` with dynamic imports inside `getStaticProps`, but I decided not to do it because it decreases code readability. We rarely touch these pages, so it should be ok to leave them as is until we update to 12.0.9.

How to test it:

- Check out this branch locally.
- Run `yarn dev`.
- Edit opened `mdx` page. It should reload without showing dialogue lightbox.

How to check that I does not break anything with these changes:

- Open any markdown page to test `remark-layouts`, they will not work at all if something is broken.
- Open any docs page to test `rehypeheaders` - if we have headers navigation on the right side - plugin works.
